src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -70,16 +73,20 @@
     public static final String TYPE = "_doc";
     public static final String QUERY_PARAM_PERIOD_START = "period_start";
     public static final String QUERY_PARAM_PERIOD_END = "period_end";
+    public static final String PARSING_ISSUE = "query_parsing";
+    public static final String GENERAL_SETTINGS = "general_settings";
+    public static final String NAME_REGEX = "[a-zA-Z0-9._-]+";
```

Comment:
Frontend limit on detector name "Detector name must contain 1-64 characters. Valid characters are a-z, A-Z, 0-9, -(hyphen) and _(underscore).", we don't allow dot `.` in detector name.

Can you verify if this regex `[a-zA-Z0-9._-]+` will allow dot `.` in detector name?

Reply:
Good catch! This regex currently allows dot `.` in detector name. On the frontend we also currently allow dot `.` in detector name(https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/913be1b6457c92e1d2bb74e605afb31882652f7b/public/utils/constants.ts#L84). Based on message we give user this a bug we have on frontend also then. I missed this as I was referring to frontend regex to make sure my backend regex was correct. I will change this to not include dot `.` and also correspondingly change this on frontend on validation PR there.

Reply:
I see you have moved the detector name regex check to validate/create/update steps, that looks good to me. So if user has old detector created with invalid name via AD API, we can show these detectors on AD dashboard correctly but will throw exception when user update it.

Reply:
Yes exactly.